src/engines/ngspice-analysis.c

```diff
@@ -933,15 +937,15 @@ static ThreadPipe *parse_noise_analysis (NgspiceAnalysisResources *resources)
 		inoise_psd = 0.;
 		onoise_psd = 0.;
 		splitted = g_regex_split_simple("[\t]*([0-9]*)[\t]*([\\+\\-]{0,1}[0-9]\\.[0-9]*[Ee][\\+\\-][0-9]*)\n",
-							&fbuffer, 0, 0);
+							fbuffer, 0, 0);
 		if (splitted[1]) {
 			if (*splitted[1]) {
 				prev_seq = seq;
 				seq = g_ascii_strtod(splitted[1], NULL);
 
 				// simple sequence sanity check
 				if (seq != prev_seq + 1)
-					return pipe;
+					break;
 
```

Comment:
use `g_autofree` for `fbuffer` in case it really has to be allocated dynamically